osquery/tables/system/darwin/connected_displays.mm

```diff
@@ -133,14 +133,20 @@ QueryData genConnectedDisplays(QueryContext& context) {
         }
       }
 
-      if (NSString* rotation = [obj valueForKey:@"spdisplays_rotation"]) {
-        if ([rotation isEqualToString:@"spdisplays_supported"]) {
-          r["rotation"] = INTEGER(1);
-        } else {
-          r["rotation"] = INTEGER(0);
+      id rotation = [obj valueForKey:@"spdisplays_rotation"];
+      if (rotation) {
+        if ([rotation isKindOfClass:[NSString class]]) {
+          // rotation is supported and the display is not rotated
+          if ([rotation isEqualToString:@"spdisplays_supported"]) {
+            r["rotation"] = INTEGER(0);
+          }
+        } else if ([rotation isKindOfClass:[NSNumber class]]) {
+          // the display is rotated (90/180/270 degrees)
+          r["rotation"] = INTEGER([rotation intValue]);
         }
       } else {
-        r["rotation"] = INTEGER(0);
+        // rotation is not supported
+        r["rotation"] = INTEGER(-1);
       }
```

Comment:
This is arguably a breaking change as `rotation` used to be `0` for not supported and now it is `-1`. :thinking: However I don't think these tables (and specifically this column) are widely used as `osqueryd` crashes if this table is queried while a display is rotated.

Alternatively we could just continue to return `1` for rotation supported and `0` for not supported to avoid breaking the current behavior.

Reply:
My hunch is that this is fine to change